src/nfa/nfa.cc

```diff
@@ -1334,12 +1338,22 @@ Nfa Mata::Nfa::determinize(
         if (S.empty()) {
             break;//this should not happen assuming all sets states_to are non empty
         }
-        Mata::Nfa::Nfa::state_set_post_iterator iterator(S.ToVector(), aut);
 
-        while (iterator.has_next()) {
-            const auto symbolTargetPair = iterator.next();
-            const Symbol currentSymbol = symbolTargetPair.first;
-            const StateSet &T = symbolTargetPair.second;
+        // add moves of S to the sync ex iterator
+        for (State q: S) {
+            sei.push_back(aut.transitionrelation[q]);
```

Comment:
Maybe calling the `sei` and `sui` as `existential_iterator`/`universal_iterator` or just `ex_it` and `all_it` would make code more clear? Just suggestion.

Reply:
Ffs, even just `it` would be better than `sei`/`sui` :D

Reply:
ok